hail/src/main/scala/is/hail/types/physical/PCanonicalNDArray.scala

```diff
@@ -107,6 +111,15 @@ final case class PCanonicalNDArray(elementType: PType, nDims: Int, required: Boo
     )
   }
 
+  def makeStridesBuilder(sourceShape: PBaseStructValue, isRowMajor: Code[Boolean], mb: EmitMethodBuilder[_]): StagedRegionValueBuilder => Code[Unit] = { srvb =>
+    val shapeCodeSeq1 = (0 until nDims).map(sourceShape[Long](_).get)
+    val shapeCodeSeq2 = (0 until nDims).map(sourceShape[Long](_).get)
```

Comment:
Is there a less awkward way of using a Code in two branches? Memoization seems not right, since value is not actually consumed twice. Store is not branch aware.

Reply:
can just make it a def

Reply:
I also think this should just be defined inline in the makeNDArray emitter, we shouldn't need to change this file.